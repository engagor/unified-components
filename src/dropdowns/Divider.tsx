import React from "react";
import "@clarabridge/unified-ui-tailwind/dist/utilities.css";

export default function Divider() {
    return <hr className="border-0 border-b-1 border-solid border-primary-color-600 mb-8 mt-8" />;
};
