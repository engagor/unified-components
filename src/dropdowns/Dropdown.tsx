import React from "react";
import "@clarabridge/unified-ui-tailwind/dist/utilities.css";

export type DropdownProps = {
    isOpen: boolean;
    children?:
        | React.ReactChild
        | React.ReactChild[];
};

export default function Dropdown(props: DropdownProps) {
    return (
        <div
            aria-role="listbox"
            aria-haspopup="listbox"
            aria-live="polite"
            aria-atomic="true"
            aria-expanded={props.isOpen}
            className="box-shadow-default rounded-8 bg-white min-w-320 max-w-400 overflow-hidden"
            tabIndex={0}
        >
            {props.children}
        </div>
    );
};
