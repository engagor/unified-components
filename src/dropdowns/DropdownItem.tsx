import React, { ReactElement } from 'react';

type DropdownItem = {
    level: number;
    selected?: boolean;
    link?: string,
    callback?: (event: React.MouseEvent) => void;
    children?:
        | React.ReactChild
        | React.ReactChild[];
};

export default function DropdownItem(props: DropdownItem): ReactElement {
    const listItemBaseClasses = [
        'w-100-percent',
        'inline-block',
        'no-underline',
        'px-16',
        'py-8',
        'cursor-pointer',
    ];

    const listItemRegularClasses = [
        ...listItemBaseClasses,
        'text-charcoal-color-400',
        'hover:bg-primary-color-200',
    ];

    const listItemActiveClasses = [
        ...listItemBaseClasses,
        'text-white',
        'bg-primary-color-600',
        'hover:bg-primary-color-200',
        'hover:text-charcoal-color-400',
    ];

    return (
        <div role="option" aria-selected={props.selected} tabIndex={0}>
            {props.link ?
                <a
                    href={props.link}
                    onClick={props.callback ?? undefined}
                    className={props.selected
                        ? listItemActiveClasses.join(' ')
                        : listItemRegularClasses.join(' ')}
                >
                    <span className={`flex pl-${props.level * 24}`}>
                        {props.children}
                    </span>
                </a> :
                <div
                    onClick={props.callback ?? undefined}
                    className={props.selected
                        ? listItemActiveClasses.join(' ')
                        : listItemRegularClasses.join(' ')}
                >
                    <span className={`flex pl-${props.level * 24}`}>
                        {props.children}
                    </span>
                </div>
            }
        </div>
    );
}
