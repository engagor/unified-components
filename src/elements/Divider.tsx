import React, { ReactElement } from 'react';

export type DividerProps = {
    color?: string;
    className?: string;
};

export default function Divider(props: DividerProps): ReactElement {
    return <hr className={`border-0 border-b-1 border-solid ${props.color ?? 'border-charcoal-color-200'} ${props.className ?? ''}`} />
}
