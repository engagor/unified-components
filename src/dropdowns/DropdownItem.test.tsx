import { renderToStaticMarkup } from 'react-dom/server';

jest.mock(
    'react-dom',
    () => {
        return {
            createPortal: (element: ReactElement, node: HTMLElement) => {
                node.innerHTML = renderToStaticMarkup(element);
            },
        }
    }
);

import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';
import DropdownItem from './DropdownItem';
import Divider from './Divider';

it('renders correctly', async () => {
    const element = document.createElement('div');

    const dropdown = await renderer
        .create(<Dropdown isOpen={true}>
            <DropdownItem level={0}>Test 1</DropdownItem>
            <DropdownItem level={1}>Test 1</DropdownItem>
            <DropdownItem level={0}>Test 3</DropdownItem>
            <DropdownItem level={0}>Test 4</DropdownItem>
            <Divider/>
            <DropdownItem level={0}>Test 5</DropdownItem>
        </Dropdown>)
        .toJSON();

    await expect(dropdown).toMatchSnapshot();
    expect(element).toMatchSnapshot();
});
