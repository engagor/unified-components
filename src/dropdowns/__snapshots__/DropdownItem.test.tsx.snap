// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-atomic="true"
  aria-expanded={true}
  aria-haspopup="listbox"
  aria-live="polite"
  className="box-shadow-default rounded-8 bg-white min-w-320 max-w-400 overflow-hidden"
  tabIndex={0}
>
  <div
    role="option"
    tabIndex={0}
  >
    <div
      className="w-100-percent inline-block no-underline px-16 py-8 cursor-pointer text-charcoal-color-400 hover:bg-primary-color-200"
    >
      <span
        className="flex pl-0"
      >
        Test 1
      </span>
    </div>
  </div>
  <div
    role="option"
    tabIndex={0}
  >
    <div
      className="w-100-percent inline-block no-underline px-16 py-8 cursor-pointer text-charcoal-color-400 hover:bg-primary-color-200"
    >
      <span
        className="flex pl-24"
      >
        Test 1
      </span>
    </div>
  </div>
  <div
    role="option"
    tabIndex={0}
  >
    <div
      className="w-100-percent inline-block no-underline px-16 py-8 cursor-pointer text-charcoal-color-400 hover:bg-primary-color-200"
    >
      <span
        className="flex pl-0"
      >
        Test 3
      </span>
    </div>
  </div>
  <div
    role="option"
    tabIndex={0}
  >
    <div
      className="w-100-percent inline-block no-underline px-16 py-8 cursor-pointer text-charcoal-color-400 hover:bg-primary-color-200"
    >
      <span
        className="flex pl-0"
      >
        Test 4
      </span>
    </div>
  </div>
  <hr
    className="border-0 border-b-1 border-solid border-primary-color-600 mb-8 mt-8"
  />
  <div
    role="option"
    tabIndex={0}
  >
    <div
      className="w-100-percent inline-block no-underline px-16 py-8 cursor-pointer text-charcoal-color-400 hover:bg-primary-color-200"
    >
      <span
        className="flex pl-0"
      >
        Test 5
      </span>
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `<div />`;
